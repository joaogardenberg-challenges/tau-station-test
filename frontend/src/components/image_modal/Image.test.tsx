import { render, screen } from '@testing-library/react'
import ThemeProvider from 'providers/Theme'

const useParams = jest.fn()
const useSelector = jest.fn()
const useQuery = jest.fn()

jest.doMock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams
}))

jest.doMock('react-redux', () => ({ useSelector }))
jest.doMock('hooks/useQuery', () => useQuery)

const Image = require('./Image').default

const renderWithTheme = (ui: any, options: any = {}) =>
  render(ui, { wrapper: ThemeProvider, ...options })

const image = {
  id: 1,
  urls: { 400: 'http://url400.com' },
  meta: { keywords: 'keywords' }
}

describe('Image', () => {
  beforeEach(() => {
    useParams.mockReturnValue({ id: 1 })
    useSelector.mockReturnValue(image)
    useQuery.mockReturnValue({ get: () => '400', set: jest.fn() })
  })

  it('renders successfully', () => {
    const { container } = renderWithTheme(<Image />)
    expect(container).toMatchSnapshot()
  })

  it('has the right attributes', () => {
    renderWithTheme(<Image />)

    const link = screen.getByRole('link')
    const img = screen.getByTestId('image')

    expect(link).toHaveAttribute('href', image.urls[400])
    expect(img).toHaveAttribute('src', image.urls[400])
    expect(img).toHaveAttribute('alt', image.meta.keywords)
  })
})
