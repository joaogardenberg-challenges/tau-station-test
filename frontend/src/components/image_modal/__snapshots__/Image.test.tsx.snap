// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders successfully 1`] = `
<div>
  <div
    class="sc-bdfBQB lfCfsQ"
  >
    <a
      class="sc-gsTEea hfGhzp"
      href="http://url400.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="keywords"
        class="loading"
        data-testid="image"
        src="http://url400.com"
      />
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-y3raj0-MuiCircularProgress-root"
        role="progressbar"
        style="width: 45px; height: 45px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;
